contracts/RequestFactory.sol

```diff
@@ -58,89 +66,110 @@ contract RequestFactory is RequestFactoryInterface {
-                                   uint[11] uintArgs,
-                                   bytes callData,
-                                   uint endowment) returns (bool[7]) {
+    function validateRequestParams(
+        address[3]  _addressArgs,
+        uint[11]    _uintArgs,
+        bytes       _callData,
+        uint        _endowment
+    )
+        public returns (bool[6])
+    {
         return RequestLib.validate(
             [
                 msg.sender,      // meta.createdBy
-                addressArgs[0],  // meta.owner
-                addressArgs[1],  // paymentData.donationBenefactor
-                addressArgs[2]   // txnData.toAddress
+                _addressArgs[0],  // meta.owner
+                _addressArgs[1],  // paymentData.donationBenefactor
+                _addressArgs[2]   // txnData.toAddress
             ],
-            uintArgs,
-            callData,
-            endowment
+            _uintArgs,
+            _callData,
+            _endowment
         );
     }
 
-    /*
-     *  The same as createRequest except that it requires validation prior to
-     *  creation.
-     *
-     *  Parameters are the same as `createRequest`
-     */
-    function createValidatedRequest(address[3] addressArgs,
-                                    uint[11] uintArgs,
-                                    bytes callData) returns (address) {
-        var is_valid = validateRequestParams(addressArgs,
-                                             uintArgs,
-                                             callData,
-                                             msg.value);
-
-        if (!is_valid.all()) {
-            if (!is_valid[0]) ValidationError(uint8(Errors.InsufficientEndowment));
-            if (!is_valid[1]) ValidationError(uint8(Errors.ReservedWindowBiggerThanExecutionWindow));
-            if (!is_valid[2]) ValidationError(uint8(Errors.InvalidTemporalUnit));
-            if (!is_valid[3]) ValidationError(uint8(Errors.ExecutionWindowTooSoon));
-            if (!is_valid[4]) ValidationError(uint8(Errors.InvalidRequiredStackDepth));
-            if (!is_valid[5]) ValidationError(uint8(Errors.CallGasTooHigh));
-            if (!is_valid[6]) ValidationError(uint8(Errors.EmptyToAddress));
-
-            // Try to return the ether sent with the message.  If this failed
-            // then throw to force it to be returned.
-            if (msg.sender.sendOrThrow(msg.value)) {
-                return 0x0;
-            }
-            throw;
-        }
-
-        return createRequest(addressArgs, uintArgs, callData);
-    }
-
+    /// Mapping to hold known requests.
     mapping (address => bool) requests;
 
-    function isKnownRequest(address _address) returns (bool) {
-        // TODO: decide whether this should be a local mapping or from tracker.
-        return requests[_address];
+    function isKnownRequest(address _address) 
+        public view returns (bool isKnown)
+    {
+        isKnown = requests[_address];
```

Comment:
*nitpick*

can you add an explicit `return` statement.  I know it's not required, but it bugs me.

Reply:
No problem